### ISSUE TITLE
auto-create database for couchdb if database is missing

### ISSUE THREAD

**Issue by Marak** [bigcompany/resource#33]
when resource connects to a couch with a specified database, if that database is not found in the database, resource should create that database and then perform the desired request.

**Comment by JoshMcCullough** [comment]
I agree that this should be a configuration option. Likely one we'd opt-in for.
